### PR TITLE
Implementing generator

### PR DESCRIPTION
**Pull request by sijinkim**
# Generator class 구현

### CODE REVIEW

File: pggan/models/modules.py [hunk 1]
```diff
@@ -5,78 +5,111 @@
...
+
+    def forward(self, x) -> torch.Tensor:
+        if self.method == 'L2':
+            order = 2
+        elif self.method == 'LC':
+            order = x.size(1)
+        else:
+            raise ValueError
+
+        return x.div(x.norm(order, dim=1, keepdim=True) + 1e-8)
```
Comment (helloybz) [inline pggan/models/modules.py]: x.norm() 후에 epsilon을 더해준 것은 본 논문에서 소개한 방법과 다르다고 생각합니다.

예를 들어 order가 2인 경우에, 논문에서는:
-  (x,y)의 pixel에 대해서 모든 feature map의 값의 제곱의 평균을 구하고
-  epsilon을 더한 후
- sqrt를 계산합니다.

이 코드는,
 - (x,y)의 pixel에 대해서 모든 feature map의 값의 제곱의 평균을 구하고 (여기까지는 논문과 같음)
 - sqrt를 계산한 후,
 - epsilon으로 나눕니다.

```suggestion
        return x.div(x.pow(2).mean(dim=1, keepdim=True).add(1e-8).sqrt())
```

File: tests/test_modules.py [hunk 2]
```diff
@@ -6,17 +6,62 @@
...
+            B, C, H, W = 3, 512, 1, 1
+            x = torch.randn(B, C, H, W)
+
+            for res in resolutions:
+                generator.set_output_image_size(res)
+                result = generator(x)
+                self.assertEqual(
+                    result.shape, (B, 3, res, res))
+
+    def test_pixel_wise_norm_method(self):
```
Comment (helloybz) [inline tests/test_modules.py]: 의미를 명확히 하기 위해 normalization이라고 표기하는 것이 좋겠습니다.

```suggestion
    def test_pixelwise_normalization_method(self):
```

File: tests/test_modules.py [hunk 3]
```diff
@@ -6,17 +6,62 @@
...
+        with torch.no_grad():
+            generator = Generator()
+            generator.fade_in_weight = 0.2
+            generator.pixel_wise_norm_method = 'LC'
```
Comment (helloybz) [inline tests/test_modules.py]: pixel_wise_norm_method는 Generator 생성자의 parameter이므로 인스턴스를 생성할 때 설정해주는 것이 좋을 것 같습니다.

```suggestion
            generator = Generator(
                pixel_wise_norm_method = 'LC',
            )
            generator.fade_in_weight = 0.2
```

File: pggan/models/modules.py [hunk 4]
```diff
@@ -5,78 +5,111 @@
...
-        return x
+        method: str = 'L2'
+    ) -> None:
+        super(PixelwiseNorm, self).__init__()
+        self.method = method
+
+    def forward(self, x) -> torch.Tensor:
+        if self.method == 'L2':
+            order = 2
+        elif self.method == 'LC':
```
Comment (helloybz) [inline pggan/models/modules.py:107]: 'LC'에 관한 내용을 논문에서 찾을 수 없습니다.
'LC'의 내용과 목적을 주석으로 설명을 추가하면 좋을 것 같습니다.

**Review by helloybz — CHANGES_REQUESTED** [review]

**Review by helloybz — APPROVED** [review]
